src/fragments/lib/storage/android/transfer-acceleration.mdx

```diff
@@ -0,0 +1,109 @@
+
+## Override storage resource
+
+Start by overriding your storage resources to enable Transfer Acceleration on your S3 bucket.
+
+```sh
+$ amplify override storage
+✅ Successfully generated "override.ts" folder at <project>/amplify/backend/storage/accelerated-bucket
+✔ Do you want to edit override.ts file now? (Y/n) · yes
+Edit the file in your editor: <project>/amplify/backend/storage/accelerated-bucket/override.ts
+```
+
+In the generated `override.ts` file use the following CDK snippet to enable transfer acceleration.
+
+```typescript
+// amplify/backend/storage/accelerated-bucket/override.ts
+import { AmplifyS3ResourceTemplate } from '@aws-amplify/cli-extensibility-helper';
+
+export function override(resources: AmplifyS3ResourceTemplate) {
+    resources.s3Bucket.accelerateConfiguration = {
+        accelerationStatus: 'Enabled'
+    }
+}
+```
+
+Next, deploy this storage resource:
+
+```sh
+amplify push
+```
+
+
+
+## Upload files using the accelerated S3 endpoint
+
+We switch to the accelerated S3 endpoint by using the `useAccelerateEndpoint` parameter set to `true` in the `AWSS3StorageUploadFileOptions`.
+<BlockSwitcher>
+
+<Block name="Java">
+
+```java
+AWSS3StorageUploadFileOptions awsS3StorageUploadFileOptions =
+              AWSS3StorageUploadFileOptions.builder().setUseAccelerateEndpoint(true).build();
+ Amplify.Storage.uploadFile(
+   "KEY",
+   file
+   awsS3StorageUploadFileOptions,
+   result -> Log.i("MyAmplifyApp", "Successfully uploaded: " + result.getKey()),
+   storageFailure -> Log.e("MyAmplifyApp", "Upload failed", storageFailure)
+);
+```
+</Block>
+<Block name="Kotlin - Callbacks">
+
+```kotlin
+val awsS3StorageUploadFileOptions = AWSS3StorageUploadFileOptions.builder().
+                                                  setUseAccelerateEndpoint(true).
+                                                  build();
+ Amplify.Storage.uploadFile(
+   "KEY",
+   file
+   awsS3StorageUploadFileOptions,
+   { Log.i("MyAmplifyApp", "Successfully uploaded: " + it.getKey()) },
+   { Log.e("MyAmplifyApp", "Upload failed", it) }
+);
+```
+</Block>
+<Block name="Kotlin - Coroutines">
+
+```kotlin
+val awsS3StorageUploadFileOptions = AWSS3StorageUploadFileOptions.builder().
+                                                    setUseAccelerateEndpoint(true).
+                                                    build();
+val upload = Amplify.Storage.uploadFile("KEY", file, awsS3StorageUploadFileOptions)
+try {
+    val result = upload.result()
+    Log.i("MyAmplifyApp", "Successfully uploaded: ${result.key}")
+} catch (error: StorageException) {
+    Log.e("MyAmplifyApp", "Upload failed", error)
+}
+);
+```
```

Comment:
nit: remove semicolons for Kotlin, not sure if this is showing it correctly
```suggestion
```kotlin
val awsS3StorageUploadFileOptions = AWSS3StorageUploadFileOptions.builder().
                                                  setUseAccelerateEndpoint(true).
                                                  build()
 Amplify.Storage.uploadFile(
   "KEY",
   file
   awsS3StorageUploadFileOptions,
   { Log.i("MyAmplifyApp", "Successfully uploaded: " + it.getKey()) },
   { Log.e("MyAmplifyApp", "Upload failed", it) }
)
```
</Block>
<Block name="Kotlin - Coroutines">

```kotlin
val awsS3StorageUploadFileOptions = AWSS3StorageUploadFileOptions.builder().
                                                    setUseAccelerateEndpoint(true).
                                                    build()
val upload = Amplify.Storage.uploadFile("KEY", file, awsS3StorageUploadFileOptions)
try {
    val result = upload.result()
    Log.i("MyAmplifyApp", "Successfully uploaded: ${result.key}")
} catch (error: StorageException) {
    Log.e("MyAmplifyApp", "Upload failed", error)
}
)
```
```